dsnp-wallet/dsnp-wallet/Source/Util/SharedImagePicker.swift

```diff
@@ -0,0 +1,84 @@
+//
```

Comment:
There is some coupling in this object that I think could be better done externally. We pass the parent to present the controllers, which I think works. But we then pass an imageView and a saveBlock. These get a bit iffy. 

The imageView locks the picker into assigning an image directly. The issue here, is that different owners might want to handle the image in different ways. For example, if we decide we want to downsize the image, this logic would require that the downsize happens in this object before assigning to the imageView. This object, more simply, should allow the user to pick a piece of content (take photo/choose photo), then pass that selected content to the owner, so the owner can do what it likes with it. That keeps things very simple in this class.

The saveBlock is also a bit iffy, because we're giving this object the concept of a save block which is vague. When we pick an image, we don't always need a save completion block associated with it. Perhaps we need a didPick delegate, but that could be combined with the above change. By passing the selected content to the owner, we can update any UI on the owner side accordingly (changing color of save button). 

By keeping these objects simple, we are more likely to re-use them.

Reply:
Thanks, that feels a lot better.